File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) { // bfs
        if (!root) return vector<int>(0);
        queue<TreeNode*> qn; qn.push(root);
        vector<int> R;
        while(!qn.empty()) {
            int size = qn.size();
            for (int i = 0; i < size; i++){
                TreeNode* N = qn.front(); qn.pop();
                if (N->left) qn.push(N->left);
                if (N->right) qn.push(N->right);
                if (i == size - 1)
                    R.push_back(N->val);
            }
        }
        return R;
    }
};

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        # BFS with queue
        if not root:
            return None
        queue = deque([root])
        while queue:
            curr = queue.popleft()
            # add left and right and figure out how to swap
            if curr.left:
                queue.append(curr.left)
            if curr.right:
                queue.append(curr.right)
            curr.left, curr.right = curr.right, curr.left
        return root
